packages/react-scripts/config/webpack.config.prod.js

```diff
@@ -53,6 +56,18 @@ const extractTextPluginOptions = shouldUseRelativeAssetPaths
   ? // Making sure that the publicPath goes back to to build folder.
     { publicPath: Array(cssFilename.split('/').length).join('../') }
   : {};
+// Check if vendor file exists
+const checkIfVendorFileExists = fs.existsSync(paths.appVendorJs);
+// If the vendor file exists, add an entry point for vendor,
+// and a seperate entry for polyfills and app index file,
+// otherwise keep only polyfills and app index.
+const appEntryFiles = [require.resolve('./polyfills'), paths.appIndexJs];
```

Comment:
@viankakrisna yup makes sense. external libs that depends on promise might get this issue. Plus polyfills are vendor libs.
can't we simply do
```js
const entryFiles = checkIfVendorFileExists
  ? {
      vendor: [require.resolve('./polyfills'), paths.appVendorJs],
      main: paths.appIndexJs,
    }
  : [require.resolve('./polyfills'), paths.appIndexJs];
```